bindata/network-diagnostics/001-rbac.yaml

```diff
@@ -0,0 +1,77 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: network-diagnostics
+  namespace: openshift-network-diagnostics
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: network-diagnostics
+rules:
+- apiGroups: [""]
+  resources:
+  - endpoints
+  - namespaces
+  - pods
+  - services
+  - nodes
+  - secrets
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups: [""]
+  resources:
+  - events
+  verbs:
+  - get
+  - list
+  - watch
+  - create
+- apiGroups: ["apiextensions.k8s.io"]
+  resources:
+  - customresourcedefinitions
+  verbs:
+    - get
+    - list
+    - watch
+- apiGroups: ['controlplane.operator.openshift.io']
+  resources: ['podnetworkconnectivitychecks']
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups: ['controlplane.operator.openshift.io']
+  resources: ['podnetworkconnectivitychecks/status']
+  verbs: ['update']
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: network-diagnostics
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: network-diagnostics
+subjects:
+- kind: ServiceAccount
+  name: network-diagnostics
+  namespace: openshift-network-diagnostics
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: system:openshift:controller:network-diagnostics-check-endpoints
```

Comment:
It's not a typo. I pulled down this code from kube-apiserver operator repo since we will end up owning all this:

https://github.com/openshift/cluster-kube-apiserver-operator/blob/0a69c78cea511a8c8e70b3910c637ffaf2a1dc36/bindata/v4.1.0/kube-apiserver/check-endpoints-rolebinding.yaml

But I don't care about var naming or argument naming bikeshedding, I can change.

Reply:
Actually I take that back, it seems it's a Kubernetes convention:

https://kubernetes.io/docs/reference/access-authn-authz/rbac/

Reply:
right, that's a way of fully-qualifying references to subjects. that's not what you're doing here - just naming a RoleBinding. This should just be a normal name.

Reply:
Right, I cargo culted what apiserver folks put as RoleBinding name as schem per https://github.com/openshift/cluster-kube-apiserver-operator/blob/0a69c78cea511a8c8e70b3910c637ffaf2a1dc36/bindata/v4.1.0/kube-apiserver/check-endpoints-rolebinding.yaml#L4 , so you confirm that scheme is not usual?

Reply:
There's no need for us to name things in that manner. This is a private role for our operator's internal use. A simple, descriptive name is sufficient.

Reply:
done